### PR TITLE
Wire nyholm/psr7 as default factory for PSR-7

### PR DESCRIPTION
**Pull request by nicolas-grekas**

### CODE REVIEW

File: DependencyInjection/SensioFrameworkExtraExtension.php [hunk 1]
```diff
@@ -148,8 +152,13 @@ public function load(array $configs, ContainerBuilder $container)
         if ($config['psr_message']['enabled']) {
             $loader->load('psr7.xml');
 
-            if (!class_exists(ServerRequestFactory::class)) {
-                $definitionsToRemove[] = 'sensio_framework_extra.psr7.argument_value_resolver.server_request';
+            if (!interface_exists(StreamFactoryInterface::class)) {
+                throw new LogicException('PSR-7 support cannot be enabled as the required dependencies are not installed. Try running "composer require nyholm/psr7".');
+            }
+
+            if (class_exists(ServerRequestFactory::class) && !class_exists(Psr17Factory::class)) {
```
Comment (fabpot) [inline DependencyInjection/SensioFrameworkExtraExtension.php]: I would remove this block

Comment (fabpot) [reply DependencyInjection/SensioFrameworkExtraExtension.php]: And everything related to Diactorios to have only Nyholm/PSR-7 as a dev req.

Comment (nicolas-grekas) [reply DependencyInjection/SensioFrameworkExtraExtension.php]: If we do that, we need to bump the minimum PHP version to 7.1. Should we do it?

Comment (fabpot) [reply DependencyInjection/SensioFrameworkExtraExtension.php]: Let's do it (and let's bump the version to a new minor)

Comment (nicolas-grekas) [reply DependencyInjection/SensioFrameworkExtraExtension.php]: Just while it was green :)
https://travis-ci.org/sensiolabs/SensioFrameworkExtraBundle/builds/504894762
pushed, let's see if it's still green with PHP 7.3 in

File: Resources/config/psr7.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
-        <service id="sensio_framework_extra.psr7.http_message_factory" class="Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory" public="false" />
+        <service id="sensio_framework_extra.psr7.http_message_factory" class="Symfony\Bridge\PsrHttpMessage\Factory\PsrHttpFactory" public="false" autowire="true" />
```
Comment (nicolas-grekas) [inline Resources/config/psr7.xml]: autowire="true", that's the trick to create decoupled configuration

Comment (nicolas-grekas) [reply Resources/config/psr7.xml]: removed: explicit aliases to the rescue

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**